keanu-project/src/main/java/io/improbable/keanu/vertices/generic/nonprobabilistic/operators/binary/BinaryOpVertex.java

```diff
@@ -1,10 +1,11 @@
 package io.improbable.keanu.vertices.generic.nonprobabilistic.operators.binary;
 
 import io.improbable.keanu.vertices.NonProbabilistic;
+import io.improbable.keanu.vertices.NonSaveableVertex;
 import io.improbable.keanu.vertices.Vertex;
 import io.improbable.keanu.vertices.dbl.KeanuRandom;
 
-public abstract class BinaryOpVertex<A, B, C> extends Vertex<C> implements NonProbabilistic<C> {
+public abstract class BinaryOpVertex<A, B, C> extends Vertex<C> implements NonProbabilistic<C>, NonSaveableVertex {
```

Comment:
Probably best not to put this on an abstract class and better to stick it on the concrete subclasses?

Reply:
Agreed, have fixed.